src/lib.rs

```diff
@@ -51,8 +51,35 @@ pub trait Color {
 /// or [`BgDynColorDisplay`](BgDynColorDisplay). If your color will be known at compile time it
 /// is recommended you avoid this.
 pub trait DynColor {
-    fn fmt_ansi_fg(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result;
-    fn fmt_ansi_bg(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result;
+
+    // XXX needed for dyn_styles.rs. Should we better return `Result`? It doesn't seem necessary right now, but maybe in the future.
+    fn get_fg(&self) -> DynStylesColor;
+    fn get_bg(&self) -> DynStylesColor;
+
+    // XXX Do you have any idea how to handle this without `DynStylesColor`? I tried half a day, but I couldn't find anything better. Without `dynamic::Rgb` we could use a `&'static str` easily, but when I hit `Rgb` everything because a pain because there wasn't an easy way to format the RGB string. Generating and returning a UTF-8 byte array, instead of `DynStylesColor`, and using `str::from_utf8(&byte_array).unwrap()` was the best I found (at least I think, this should work). But that would make everything else a little less efficient and more complex, so I went with `DynStylesColor`
+    fn fmt_ansi_fg(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let DynStylesColor::Ansi(ansi) = self.get_fg() {
+            write!(f, "{}", ansi)
+        } else {
+            unreachable!() // RGB based implementations implement this themself
+        }
+    }
+
+    // XXX If the `if let` costs too much, I can restore the previous version. I made this change before I realized `Rgb` will give me such troubles. I let it as it is now because it's less code and the `if let` shouldn't use too much ressources
+    fn fmt_ansi_bg(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let DynStylesColor::Ansi(ansi) = self.get_bg() {
+            write!(f, "{}", ansi)
+        } else {
+            unreachable!() // RGB based implementations implement this themself
+        }
+    }
+}
+
+// XXX Any idea for a better name?
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum DynStylesColor {
```

Comment:
Why is Ansi an `&'static str` instead of the `AnsiColor` enum? I'm also a bit confused what the benefit of this is over the existing `DynColors` enum?

Reply:
I started with a `&'static str` as the values of colors in `Style`. I thought this would be a bit more performant. When I met `Rgb` from which I couldn't get a `&'static str`, I created DynStylesColor.

Now that I see my finished code, and with the help of your questions, I see that `&'static str` shouldn't be significant more performant than using DynColors (if at all). And that this led me to a bad design.

I think a big problem was, that I failed to step back and see if what I do still makes sense when I hit problems.